### DISCUSSION TITLE
input_httpserver 对influxdb协议特殊字符约束的改进建议

### DISCUSSION THREAD

**Discussion by snakorse** [alibaba/ilogtail#564] [Ideas]
## 背景
实际使用中遇到一个case: input(influxdb) -> flusher(influxdb)的场景，当输入指标名时` a.b` 这样的结构时，输出指标变成了`a_b`

## 问题
当前 `input_httpserver` 插件用到的 influxdb decoder 中，对一些特殊字符默认替换为了"_"，代码位置：
https://github.com/alibaba/ilogtail/blob/ac73089571f945ec7fbc170732ef90b9d4162c46/helper/decoder/influxdb/decoder.go#L122
但这可能是不合适的，因为在这之前，已经通过 influxdb的协议正确的解析了出了数据，说明 metrics_name 、tag_key 都是 influxdb 协议合法可接受的字符
如果因为后续处理的需要想替换某些字符，应该通过后续追加诸如 `rename` 这样的 processor 来实现，而不应该在 `input` 中默认替换，这会导致后续再 flush 到`influxdb`的时候指标名发生不必要的变化

## 方案
`input_httpserver` 插件增加一个参数选项：`StrictChars` `bool`，表示 是否开启严格模式，开启的情况下，才执行现在的`ReplaceInvalidChars`的逻辑，为保持兼容性，该参数可以默认为 `true`

**Comment by messixukejia** [comment]
看上去是可行的。 

@EvanLjp LJ 也确认下

**Comment by messixukejia** [comment]
StrictChars 跟 最近加的extendfiled 要不要融合下？ 尽量少的开关

**Reply by snakorse** [reply]
嗯 复用现在的extendfields是可以的，也可以解释的通

**Reply by messixukejia** [reply]
在你们正式上线前，extendfields也可以再考虑改个更通用的名字

**Reply by snakorse** [reply]
FieldsExtend 我感觉是可以的
如果要更通用，或许可以考虑Version字段

**Reply by messixukejia** [reply]
那就FieldsExtend吧

Version有低版本不使用的意思在，可能不是太合适
